tests/common/database_fixture.cpp

```diff
@@ -380,7 +425,30 @@ string database_fixture::generate_anon_acct_name()
    //    to workaround issue #46
    return "anon-acct-x" + std::to_string( anon_acct_count++ );
 }
+bool database_fixture::validation_current_test_name_for_setting_api_limit(string & current_test_name) const
+{
+   vector <string> valid_testcase {"api_limit_get_account_history_operations","api_limit_get_account_history"
+      ,"api_limit_get_grouped_limit_orders","api_limit_get_relative_account_history"
+      ,"api_limit_get_account_history_by_operations","api_limit_get_asset_holders"
+      ,"api_limit_get_key_references","api_limit_get_limit_orders"
+      ,"api_limit_get_call_orders","api_limit_get_settle_orders"
+      ,"api_limit_get_order_book","api_limit_lookup_accounts"
+      ,"api_limit_lookup_witness_accounts","api_limit_lookup_committee_member_accounts"
+      ,"api_limit_lookup_vote_ids","api_limit_get_account_limit_orders"
+      ,"api_limit_get_collateral_bids","api_limit_get_top_markets"
+      ,"api_limit_get_trade_history", "api_limit_get_trade_history_by_sequence"
+      ,"api_limit_get_withdraw_permissions_by_giver","api_limit_get_withdraw_permissions_by_recipient"
+      ,"api_limit_get_full_accounts2"};
```

Comment:
Although the code is moved here, I'd say it's still ugly. Ideally the arbitrary test case names should be moved out of `database_fixture` class.

Reply:
1. Please suggest where the code can be moved or any other better approach
2. The above comment was discussed by @pmconrad during the review of PR 
https://github.com/bitshares/bitshares-core/pull/1513#discussion_r259608144
Should a new ticket be made to refactor the database_fixture.cpp?
Plz suggest what needs to be done.

Reply:
I was not saying that we reject the changes this time. Actually it's a good idea to create a new issue for the code refactory, because it's not high priority (although this one isn't either).

About what's better approach, I thought that we could create a new fixture class that is derived from `database_fixture`, add a hook function there, every individual test case that uses that fixture can define their own parameters for initializing the app instance.

Reply:
Made an issue to follow up on the above:
https://github.com/bitshares/bitshares-core/issues/1907